src/http_client_async.zig

```diff
@@ -3480,21 +3480,16 @@ pub fn handleResponseMetadata(
                     } else {
                         var url_buf = URLBufferPool.get(default_allocator);
                         const original_url = this.url;
-                        const port = original_url.getPortAuto();
-
-                        if (port == original_url.getDefaultPort()) {
-                            this.url = URL.parse(std.fmt.bufPrint(
-                                &url_buf.data,
-                                "{s}://{s}{s}",
-                                .{ original_url.displayProtocol(), original_url.displayHostname(), location },
-                            ) catch return error.RedirectURLTooLong);
-                        } else {
-                            this.url = URL.parse(std.fmt.bufPrint(
-                                &url_buf.data,
-                                "{s}://{s}:{d}{s}",
-                                .{ original_url.displayProtocol(), original_url.displayHostname(), port, location },
-                            ) catch return error.RedirectURLTooLong);
-                        }
+
+                        const new_url_ = bun.JSC.URL.join(
+                            bun.String.fromUTF8(original_url.href),
+                            bun.String.fromUTF8(location),
+                        );
+                        defer new_url_.deref();
+
+                        const new_url = new_url_.byteSlice();
```

Comment:
URL.join returns a latin1 or utf-16 encoded string. It must be converted to UTF8 using the usual methods

Reply:
Fixed.